src/main/scala/scalaz/stream/Process.scala

```diff
@@ -451,8 +451,19 @@ sealed trait Process[+F[_], +O]
 
   }
 
+  final def uncons[F2[x] >: F[x], O2 >: O](implicit F: Monad[F2], C: Catchable[F2]): F2[(O2, Process[F2, O2])] =
+    unconsOption(F, C).map(_.getOrElse(throw new NoSuchElementException))
 
-
+  final def unconsOption[F2[x] >: F[x], O2 >: O](implicit F: Monad[F2], C: Catchable[F2]): F2[Option[(O2, Process[F2, O2])]] = step match {
+    case Step(head, next) => head match {
+      case Emit(as) => F.point(as.headOption.map(x => ((x, Process.emitAll[O2](as drop 1) +: next))))
```

Comment:
This is incorrect, as the logic assumes that `as` will only be empty if the stream is at an end!  That is not the case though, as `step` is free to produce an `Emit(Nil)` at any time.